programs/zstd.1.md

```diff
@@ -144,6 +144,11 @@ the last one takes effect.
     Due to the chaotic nature of dynamic adaptation, compressed result is not reproducible.
     _note_ : at the time of this writing, `--adapt` can remain stuck at low speed
     when combined with multiple worker threads (>=2).
+* `--stream-size=#` :
+    When handling input from a stream, `zstd` must guess how large the source size
+    will be when optimizing compression parameters. This option sets the pledged source
+    size of a stream to eliminate that guesswork. Note that the pledged size must be exact;
+    incorrect stream sizes will cause an error.
```

Comment:
minor comment : 
In term of presentation priority,
I would rather state first that the information must be exact because it will be part of the produced frame header.
And then, second, that the information is also used to help optimize parameters, resulting in better / faster compression.

This makes this command more specific compared to `--size-hint=#`, which lacks the first part.

Reply:
Good point, I was trying to come up with a way to differentiate the two in the docs. I'll change it.